### PR TITLE
[run_clm example] add torch_dtype option for model load.

### PR DESCRIPTION
**Pull request by sywangyi**
for BLOOM 175B model. peak memory will reduce about 350G for inference. the weight of BLOOM in model hub is bfloat16

Signed-off-by: Wang, Yi A <yi.a.wang@intel.com>

# What does this PR do?

reduce the peak memory for BLOOM inference.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

- trainer: @sgugger

### CODE REVIEW

**Comment by sywangyi** [comment]
@yao-matrix @jiqing-feng @sgugger please help to review

**Review by sgugger — COMMENTED** [review]
Thanks for your PR. If we add this to the example, then we should support other torch dtypes (the user will likely want to specify a given one) like `"float16"` or `"bfloat16"`.

**Comment by sywangyi** [comment]
@sgugger done,  add other torch dtypes per review comment

File: examples/pytorch/language-modeling/run_clm.py [hunk 1]
```diff
@@ -391,7 +392,9 @@ def main():
             cache_dir=model_args.cache_dir,
             revision=model_args.model_revision,
             use_auth_token=True if model_args.use_auth_token else None,
-            torch_dtype=model_args.torch_dtype,
+            torch_dtype=model_args.torch_dtype
+            if model_args.torch_dtype in ["auto", None]
+            else getattr(torch, model_args.torch_dtype),
```
Comment (sgugger) [inline examples/pytorch/language-modeling/run_clm.py:397]: Let's put this on its own line before this model call for readability.

**Review by sgugger — APPROVED** [review]
Thanks a lot! Just one last nit and we can merge this :-)

File: examples/pytorch/language-modeling/run_clm.py (hunk 1 above)
Comment (sywangyi) [reply examples/pytorch/language-modeling/run_clm.py]: done

**Pull request merged**